### PR TITLE
CATTY-638 Update AudioKit to Version 5.3.1

### PR DESCRIPTION
**Pull request by srinner**
Update AudioKit to Version 5.3.1 and check whether all Sound- and all Speech-Bricks are still working correctly (including volume change, mute, etc).

### Your checklist for this pull request
Please review the [contributing guidelines](https://github.com/Catrobat/Catty/blob/develop/README.md) and [wiki pages](https://github.com/Catrobat/Catroid/wiki/) of this repository.

- [x] Include the name of the Jira ticket in the PR’s title
- [x] Verify that the Jira ticket is in the status *Ready for Development*
- [x] Include a summary of the changes plus the relevant context
- [x] Choose the proper base branch (*develop*)
- [x] Confirm that the changes follow the project’s coding guidelines
- [x] Verify that the changes generate no compiler or linter warnings
- [x] Perform a self-review of the changes
- [x] Verify to commit no other files than the intentionally changed ones
- [x] Include reasonable and readable tests verifying the added or changed behavior
- [x] Confirm that new and existing unit tests pass locally
- [x] Check that the commits’ message style matches the [project’s guideline](https://github.com/Catrobat/Catroid/wiki/Commit-Message-Guidelines)
- [x] Stick to the project’s git workflow (rebase and squash your commits)
- [x] Verify that your changes do not have any conflicts with the base branch
- [ ] After the PR, verify that all CI checks have passed
- [ ] Post a message in the *#catty* [Slack channel](https://catrobat.slack.com) and ask for a code reviewer

### CODE REVIEW

**Comment by wallisch** [comment]
Could you please try to bump this to AudioKit 5.3.1? 
Also, please don't forget to update DunneAudioKit too.

File: src/Catty/PlayerEngine/AudioEngine/AudioPlayer.swift [hunk 1]
```diff
@@ -98,7 +98,7 @@ class AudioPlayer: NSDiscardableContent {
     }
 
     func isPlaying() -> Bool {
```
Comment (wallisch) [inline src/Catty/PlayerEngine/AudioEngine/AudioPlayer.swift:100]: Could you please create computed properties for `isPlaying` and `isPaused` here that switch based on the player state?
```swift
var isPlaying: Bool {
    player.status == .playing
}

var isPaused: Bool {
    player.status == .paused
}
```
And then use them everywhere instead of other methods to get the state, this makes it easier to understand.
Also remove the current isPaused property and the isPlaying() function.

File: src/CattyTests/PlayerEngine/AudioEngine/UnitTests/AudioPlayerTests.swift [hunk 2]
```diff
@@ -59,37 +59,37 @@ final class AudioPlayerTests: XCTestCase {
     func testPlayDiscardedPlayerExpectPlayerNotToPlay() {
         audioPlayer.isDiscarded = true
         audioPlayer.play(expectation: nil)
-        expect(self.audioPlayer.player.isPlaying) == false
```
Comment (wallisch) [inline src/CattyTests/PlayerEngine/AudioEngine/UnitTests/AudioPlayerTests.swift:62]: This file can be left unchanged with the previous changes

File: src/CattyTests/PlayerEngine/AudioEngine/Mocks/AudioEngineFingerprintingStub.swift [hunk 3]
```diff
@@ -36,7 +36,10 @@ class AudioEngineFingerprintingStub: Pocket_Code.AudioEngine {
             speechSynth.utteranceVolume = muted ? 0.0 : 1.0
         }
 
-        recorder = try? AudioKit.NodeRecorder(node: engineOutputMixer)
+        let tempFile: AVAudioFile? = AudioKit.NodeRecorder.createTempFile()
+        let file = try? AVAudioFile(forWriting: tempFile!.url, settings: tempFile!.fileFormat.settings)
+
+        recorder = try? AudioKit.NodeRecorder(node: engineOutputMixer, file: file)
```
Comment (wallisch) [inline src/CattyTests/PlayerEngine/AudioEngine/Mocks/AudioEngineFingerprintingStub.swift:42]: Are this changes still necessary with AK 5.3.1? Should be done automatically

**Review by wallisch — CHANGES_REQUESTED** [review]

File: src/CattyTests/PlayerEngine/AudioEngine/Mocks/AudioEngineFingerprintingStub.swift (hunk 3 above)
Comment (srinner) [reply src/CattyTests/PlayerEngine/AudioEngine/Mocks/AudioEngineFingerprintingStub.swift:42]: Yes it is necessary, otherwise the test cases fail.

File: src/Catty/PlayerEngine/AudioEngine/AudioPlayer.swift [hunk 4]
```diff
@@ -52,7 +57,7 @@ class AudioPlayer: NSDiscardableContent {
         playingQueue.sync {
             soundCompletionHandler()
             if !self.isDiscarded {
-                if player.isPlaying {
+                if player.playerNode.isPlaying {
```
Comment (wallisch) [inline src/Catty/PlayerEngine/AudioEngine/AudioPlayer.swift:60]: Use `self.isPlaying` here too please

File: src/Catty/PlayerEngine/AudioEngine/AudioPlayer.swift [hunk 5]
```diff
@@ -76,16 +81,14 @@ class AudioPlayer: NSDiscardableContent {
     }
 
     func pause() {
-        if player.isPlaying {
+        if player.playerNode.isPlaying {
```
Comment (wallisch) [inline src/Catty/PlayerEngine/AudioEngine/AudioPlayer.swift:84]: And here

**Review by wallisch — APPROVED** [review]
